sql/modules/Menu.sql

```diff
@@ -80,7 +83,14 @@ $$
                                            like c.path::text || ',%')
             GROUP BY n.position, n.id, c.level, n.label, c.path, c.positions,
                      n.parent
-            ORDER BY string_to_array(c.positions, ',')::int[]
+            ORDER BY string_to_array(c.positions, ',')::int[])
+        SELECT t.*, ch.children from t
+        LEFT JOIN (
+            SELECT parent, level, array_agg(id) as children
+            FROM t
```

Comment:
Why do we do it this way instead of querying the children by the parent's id when we retrieve them? That doesn't require the join, instead returning records from an index search.